distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/restito/RestitoRetrievalResponse.java

```diff
@@ -0,0 +1,172 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.common.test.restito;
+
+import static java.lang.Thread.sleep;
+
+import static com.xebialabs.restito.semantics.Action.composite;
+import static com.xebialabs.restito.semantics.Action.contentType;
+import static com.xebialabs.restito.semantics.Action.custom;
+import static com.xebialabs.restito.semantics.Action.header;
+
+import java.io.IOException;
+
+import org.glassfish.grizzly.http.server.Response;
+import org.slf4j.LoggerFactory;
+import org.slf4j.ext.XLogger;
+
+import com.xebialabs.restito.semantics.Action;
+import com.xebialabs.restito.semantics.Function;
+
+/**
+ * Encompasses the response for a product retrieval for a Restito stub server.
+ * Allows for the simple setup of specific responses, slow sending, and simulated
+ * network failure retry testing.
+ */
+public class RestitoRetrievalResponse implements Function<Response, Response> {
+    private static final XLogger LOGGER = new XLogger(LoggerFactory.getLogger(
+            RestitoRetrievalResponse.class));
+
+    private String responseMessage;
+
+    private int messageDelayMs = 0;
+
+    private int numberOfFailures = 0;
+
+    private int numberOfCalls = 0;
+
+    private int numberOfRetries = 0;
+
+    /**
+     * Constructor for a response that has no delay and no planned failures.
+     *
+     * @param responseMessage Message to be sent in response.
+     */
+    public RestitoRetrievalResponse(String responseMessage) {
+        this.responseMessage = responseMessage;
+    }
+
+    /**
+     * Constructor for a slow response that has a delay and no planned failures.
+     *
+     * @param responseMessage Message to be sent in response.
+     * @param messageDelayMs  Delay in ms to wait between sending each character of the message.
+     */
+    public RestitoRetrievalResponse(String responseMessage, int messageDelayMs) {
```

Comment:
We should use `Duration` here instead of `int`, that way people can use `Duration.of(10, MILLIS)`. Or we could use `int time, TimeUnit unit` and have people do `10, MILLISECONDS`.

Wonder if we should also have a way to provide a number of chunks. That would allow people to send more than one character at a time. Thoughts?